Framework/MDAlgorithms/src/MDNorm.cpp

```diff
@@ -970,6 +972,17 @@ for (int64_t i = 0; i < ndets; i++) {
   // If the dtefctor is a group, this should be the ID of the first detector
   const auto detID = detector.getID();
 
+  // get the flux spectrum number
+  size_t wsIdx = 0;
+  if (m_diffraction) {
+    auto index = fluxDetToIdx.find(detID);
```

Comment:
`fluxDetToIdx` needs to be `const` for the above operation to be thread-safe
https://en.cppreference.com/w/cpp/container